charts/aws-cloud-controller-manager/values.yaml

```diff
@@ -105,6 +105,16 @@ resources:
     cpu: 200m
   #   memory: 300Mi
 
+# env -- Pod environment variables
+env: {}
+
+hostNetworking: false
```

Comment:
This won't work. CCM in many cases need to work before CNI and therefore must use the host's network.

Reply:
Hey @olemarkus 

please correct me if I'm wrong, but I'm only declare the current default here. I will remove it from PR, since it's out of scope.

Anyway, why CCM is needed for an CNI? https://github.com/aws/amazon-vpc-cni-k8s does not depend on CCM and  ignores any taints by default.

Please explain a case, I guess I miss one.

Reply:
See https://kubernetes.io/docs/concepts/architecture/cloud-controller/#route-controller

Its fine to have this configurable, although I don't think that serves much purpose. The point of using IRSA with CCM is typically to remove privileges from the control plane role, not protect the control plane role (which is usually the case otherwise). Thus that it runs hostNetworking is not a problem.

While the VPC CNI helm chart tolerates all taints, it is not given that other CNIs' install instructions do so. Which means that the CNI DS and CCM DS block each other.

Reply:
Thanks!

> Its fine to have this configurable, although I don't think that serves much purpose. 

It's already configurable on master, introduced in https://github.com/kubernetes/cloud-provider-aws/pull/173. The if condition leads to an undefined value by default.